### PR TITLE
306752 add visual tests for overlay

### PR DESCRIPTION
**Pull request by GerardasDilda**

### CODE REVIEW

File: e2e/src/overlay-demo.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { AppPage } from './app.po';
+import { browser, by } from 'protractor';
+import { testScreenshot } from './screenshot-util';
+
```
Comment (GabrieleGe) [inline e2e/src/overlay-demo.e2e-spec.ts]: missleading. its overlay demo.

File: e2e/src/overlay-demo.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+describe('buttons demo', () => {
+  let page: AppPage;
+
+  beforeAll(async () => {
+    page = new AppPage();
+    await page.navigateToOverlayDemo();
+  });
+
+  it('looks correct for SnackBar', async () => {
+    const bothButton = browser.element(by.css('button[aria-label="buttonForSnackBar"]'));
```
Comment (GabrieleGe) [inline e2e/src/overlay-demo.e2e-spec.ts]: snackBarButton?

File: e2e/src/overlay-demo.e2e-spec.ts (hunk 1 above)
Comment (GerardasDilda) [reply e2e/src/overlay-demo.e2e-spec.ts]: Fixed

File: e2e/src/overlay-demo.e2e-spec.ts (hunk 2 above)
Comment (GerardasDilda) [reply e2e/src/overlay-demo.e2e-spec.ts]: Fixed

**Pull request merged**